src/main/java/me/ferdz/placeableitems/init/PlaceableItemsMap.java

```diff
@@ -0,0 +1,48 @@
+package me.ferdz.placeableitems.init;
+
+import me.ferdz.placeableitems.PlaceableItems;
+import me.ferdz.placeableitems.block.PlaceableItemsBlock;
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+
+import java.util.LinkedHashMap;
+
+public class PlaceableItemsMap extends LinkedHashMap<Item, PlaceableItemsBlock> {
+
+    private static final Object lock = new Object();
+    private static PlaceableItemsMap instance;
+
+    /**
+     * Private constructor to ensure singleton pattern
+     */
+    private PlaceableItemsMap() {
+        super();
+    }
+
+    /**
+     * Gets the singleton instance with a lock to ensure thread safety.
+     * @return the only available instance of the map
+     */
+    public static PlaceableItemsMap instance() {
+        synchronized (lock) {
+            if(instance == null) instance = new PlaceableItemsMap();
+            return instance;
+        }
+    }
+
+    /**
+     * Set the registry name of the block and registers it to the Block registry.
+     * @param item the associated item
+     * @param block the resulting block
+     * @param name the registry name for this block
+     * @return the block that was just added
+     */
+    public PlaceableItemsBlock put(Item item, PlaceableItemsBlock block, String name) {
+        if(!this.containsValue(block)) {
+            block.setRegistryName(PlaceableItems.MODID, name);
+            GameRegistry.findRegistry(Block.class).register(block);
```

Comment:
Is there anyway to pass the registry we get from `onBlocksRegistry` instead of finding the instance at every register call?

Reply:
Didn't want to hold a reference to the registry, its O(1) so it shouldn't be a performance problem.

Reply:
Since you moved it to the register in the block, we can just pass the reference there